source/validate_cfg.cpp

```diff
@@ -59,56 +60,72 @@ using bb_ptr = BasicBlock*;
 using cbb_ptr = const BasicBlock*;
 using bb_iter = vector<BasicBlock*>::const_iterator;
 
-/// @brief Sorts the blocks in a CFG given the entry node
-///
-/// Returns a vector of basic block pointers in a Control Flow Graph(CFG) which
-/// are sorted in the order they were accessed in a post order traversal.
-///
-/// @param[in] entry the first block of a CFG
-/// @param[in] depth_hint a hint about the depth of the CFG
-///
-/// @return A vector of pointers in the order they were access in a post order
-/// traversal
-vector<const BasicBlock*> PostOrderSort(const BasicBlock& entry, size_t size) {
-  struct block_info {
-    cbb_ptr block;
-    bb_iter iter;
-  };
+using get_blocks_func = function<const vector<BasicBlock*>*(const BasicBlock*)>;
+
+struct block_info {
+  cbb_ptr block;
+  bb_iter iter;
+};
+
+bool IsBackEdge(vector<block_info> staged, uint32_t id) {
+  for (auto b : staged) {
+    if (b.block->get_id() == id) return true;
+  }
```

Comment:
I see what this function replaced below.  Its implementation could be
return staged.find(id) != staged.end().